src/lib/components/Swap/index.tsx

```diff
@@ -18,54 +23,50 @@ import Settings from './Settings'
 import SwapButton from './SwapButton'
 import Toolbar from './Toolbar'
 
-interface DefaultTokenAmount {
-  address?: string | { [chainId: number]: string }
-  amount?: number
-}
-
-interface SwapDefaults {
-  tokenList: string | TokenInfo[]
-  input: DefaultTokenAmount
-  output: DefaultTokenAmount
-}
-
-const DEFAULT_INPUT: DefaultTokenAmount = { address: 'ETH' }
-const DEFAULT_OUTPUT: DefaultTokenAmount = {}
-
-function useSwapDefaults(defaults: Partial<SwapDefaults> = {}): SwapDefaults {
-  const tokenList = defaults.tokenList || DEFAULT_TOKEN_LIST
-  const input: DefaultTokenAmount = defaults.input || DEFAULT_INPUT
-  const output: DefaultTokenAmount = defaults.output || DEFAULT_OUTPUT
-  input.amount = input.amount || 0
-  output.amount = output.amount || 0
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  return useMemo(() => ({ tokenList, input, output }), [])
-}
-
+export type DefaultAddress = string | { [chainId: number]: string } | 'NATIVE'
 export interface SwapProps {
+  theme?: Theme
```

Comment:
A lot of these props are already specified as part of `WidgetProps`, not in `SwapProps`. `SwapProps` should only be those things that will be specific to swapping, otherwise you break the abstraction layer between widget/swap. I think the only props should be:
- default props
- convenience fee